### PR TITLE
gtk: Implement Display for text buffers

### LINKED ISSUE #1425: Change the Display implementation for EntryBuffer
The current implementation of `Display` for `EntryBuffer` just returns `"EntryBuffer"`. Although intuitively one would expect for it to return the internal GString converted to a String.

<https://gtk-rs.org/gtk4-rs/stable/latest/docs/src/gtk4/auto/entry_buffer.rs.html#248-252>

### PR DESCRIPTION
**Pull request by AbuShawarib**
Implement Display trait for GtkTextBuffer and GtkEntryBuffer.

Closes #1425

### CODE REVIEW

File: gtk4/src/text_buffer.rs [hunk 1]
```diff
@@ -103,3 +103,9 @@ pub trait TextBufferExtManual: sealed::Sealed + IsA<TextBuffer> + 'static {
 }
 
 impl<O: IsA<TextBuffer>> TextBufferExtManual for O {}
+
+impl fmt::Display for TextBuffer {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str(self.text(&self.start_iter(), &self.end_iter(), true).as_str())
```
Comment (bilelmoussaoui) [inline gtk4/src/text_buffer.rs:109]:
```suggestion
        f.write_str(&self.text(&self.start_iter(), &self.end_iter(), true))
```

File: gtk4/src/entry_buffer.rs [hunk 2]
```diff
@@ -127,3 +128,9 @@ impl Default for EntryBuffer {
         glib::Object::new()
     }
 }
+
+impl fmt::Display for EntryBuffer {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str(self.text().as_str())
```
Comment (bilelmoussaoui) [inline gtk4/src/entry_buffer.rs:134]:
```suggestion
        f.write_str(&self.text())
```

**Review by bilelmoussaoui — COMMENTED** [review]
otherwise lgtm

**Comment by bilelmoussaoui** [comment]
The CI will fail as we have to switch to 1.70

**Comment by AbuShawarib** [comment]
@LevitatingBusinessMan maintainers changed their minds on implementing Display:

> you would get different results from the Display impl depending on what type you have the object in. as glib::Object it's going to print something else.

**Pull request closed without merging**